### PR TITLE
add lint git hook

### PR DESCRIPTION
**Pull request by spencer-codaio**
mostly takes files + process from the `coda` repo

image
triggers on push now

@coda/packs

### CODE REVIEW

**Comment by huayang-codaio** [comment]
Oh I recall why I didn't proceed earlier because I was using a npm to run hooks but we didn't want to propagate the npm dependency to downstream. Using a local script works.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.circleci/*
+.githooks/*
+.idea/*
+.vscode/*
+node_modules/*
+./documentation/snippets
+./documentation/examples
```
Comment (jonathan-codaio) [inline .eslintignore:7]: We definitely want these to get linted, we just want them to have their own rule overrides. I'm trying to understand, is the custom .eslintrc.js not working properly or something?

File: documentation/.eslintrc.js [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
-        'max-len': ['error', { 'code': 80 }],
-        'quotes': ['error', 'double'],
+        '@typescript-eslint/no-unused-vars': [
+          'error',
+          {
+            varsIgnorePattern: 'response|datasourceUrl|MySchema',
+            argsIgnorePattern: 'context|param',
+          },
+        ],
+        'max-len': ['error', {code: 80}],
```
Comment (huayang-codaio) [inline documentation/.eslintrc.js:15]: 80 is too small. can we do 

```
    'max-len': [
      'error',
      {
        code: 120,
        tabWidth: 2,
        ignoreUrls: true,
        ignoreStrings: true,
        ignoreTemplateLiterals: true,
        ignoreRegExpLiterals: true,
        ignorePattern: '^import ',
      },
    ],

```

**Review by huayang-codaio — APPROVED** [review]
i have no other comments.

File: documentation/.eslintrc.js (hunk 2 above)
Comment (spencer-codaio) [reply documentation/.eslintrc.js:15]: just to make sure, this is for the `documentation` linter that you want to make the change for? cc @ekoleda-codaio

Comment (huayang-codaio) [reply documentation/.eslintrc.js:15]: oops mb. meant for the main one. and the main one has no configure for max-len. new configurations can be added now or later. I don't have a preference.

**Pull request merged**